OPEN_PROBLEMS/ENHANCED_BITSWAP_GRAPHSYNC.md

```diff
@@ -0,0 +1,128 @@
+# Enhanced Bitswap/GraphSync with more Network Smarts
+
+## Short Description
+> In one sentence or paragraph.
+
+Once IPFS has resolved the CID to a peerID and further to the exact location of the node that  stores the requested content, it then uses a very simple protocol called bitswap to exchange content between the requestor and the server.
```

Comment:
This is broadly correct, however Bitswap also participates in the discovery phase:
- IPFS asks Bitswap for the blocks corresponding to a set of CIDs
- Bitswap sends a request for the CIDs to all connected peers
- If no connected peers have a particular block, Bitswap searches for more peers by querying the DHT for the first CID (usually the root node in a DAG)